internal/matchers/matchers.go

```diff
@@ -126,20 +129,17 @@ func (m *Matcher) calculateDiff(actual interface{}) ([]byte, error) {
 	}
 
 	// Use a mergePatch to produce a diff between the two objects.
-	originalWithModifiedJSON, err := jsonpatch.MergePatch(originalJSON, actualJSON)
-	if err != nil {
-		return nil, err
-	}
-	rawDiff, err := jsonpatch.CreateMergePatch(originalJSON, originalWithModifiedJSON)
```

Comment:
Sorry I did not noticed this before, this is two-ways merge, but we need a plain merge patch like proposed in this change. Good catch